From 5a8a2f81ef8650f06d1d9d268add612ab46025f1 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 12 Jun 2018 22:57:04 -0700
Subject: [PATCH] Remove man files which cant be built

Remove need for pandoc-prebuilt during cross build

Fixes errors

| CMake Error at libibumad/man/cmake_install.cmake:105 (file):
|   file INSTALL cannot find
|   "/mnt/a/oe/build/tmp/work/core2-64-bec-linux-musl/rdma-core/18.1-r0/git/buildlib/pandoc-prebuilt/41bbb0bed7a781be59e8c0dcd8b7278af2ce6882".
| Call Stack (most recent call first):
|   cmake_install.cmake:48 (include)

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Update for version 26.1. Remove more man files which can no be built.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 infiniband-diags/man/CMakeLists.txt | 34 -------------------
 libibumad/man/CMakeLists.txt        |  2 --
 libibverbs/man/CMakeLists.txt       | 52 -----------------------------
 librdmacm/man/CMakeLists.txt        |  2 --
 providers/efa/man/CMakeLists.txt    |  2 --
 providers/mlx4/man/CMakeLists.txt   |  1 -
 providers/mlx5/man/CMakeLists.txt   | 65 -------------------------------------
 7 files changed, 158 deletions(-)

diff --git a/infiniband-diags/man/CMakeLists.txt b/infiniband-diags/man/CMakeLists.txt
index 1f1f91c9..094a389f 100644
--- a/infiniband-diags/man/CMakeLists.txt
+++ b/infiniband-diags/man/CMakeLists.txt
@@ -46,40 +46,6 @@ rdma_rst_common(
 )
 
 rdma_man_pages(
-  check_lft_balance.8.in.rst
-  dump_fts.8.in.rst
-  ibaddr.8.in.rst
-  ibcacheedit.8.in.rst
-  ibccconfig.8.in.rst
-  ibccquery.8.in.rst
-  ibfindnodesusing.8.in.rst
-  ibhosts.8.in.rst
-  ibidsverify.8.in.rst
-  iblinkinfo.8.in.rst
-  ibnetdiscover.8.in.rst
-  ibnodes.8.in.rst
-  ibping.8.in.rst
-  ibportstate.8.in.rst
-  ibqueryerrors.8.in.rst
-  ibroute.8.in.rst
-  ibrouters.8.in.rst
-  ibstat.8.in.rst
-  ibstatus.8.in.rst
-  ibswitches.8.in.rst
-  ibsysstat.8.in.rst
-  ibtracert.8.in.rst
-  infiniband-diags.8.in.rst
-  perfquery.8.in.rst
-  saquery.8.in.rst
-  sminfo.8.in.rst
-  smpdump.8.in.rst
-  smpquery.8.in.rst
-  vendstat.8.in.rst
-  )
-
-rdma_alias_man_pages(
-  dump_fts.8 dump_lfts.8
-  dump_fts.8 dump_mfts.8
   )
 
 if (ENABLE_IBDIAGS_COMPAT)
diff --git a/libibumad/man/CMakeLists.txt b/libibumad/man/CMakeLists.txt
index 185584a0..857a4b8b 100644
--- a/libibumad/man/CMakeLists.txt
+++ b/libibumad/man/CMakeLists.txt
@@ -15,7 +15,6 @@ rdma_man_pages(
   umad_get_mad_addr.3
   umad_get_pkey.3
   umad_get_port.3
-  umad_init.3.md
   umad_open_port.3
   umad_poll.3
   umad_recv.3
@@ -38,5 +37,4 @@ rdma_alias_man_pages(
   umad_class_str.3 umad_method_str.3
   umad_get_ca.3 umad_release_ca.3
   umad_get_port.3 umad_release_port.3
-  umad_init.3 umad_done.3
   )
diff --git a/libibverbs/man/CMakeLists.txt b/libibverbs/man/CMakeLists.txt
index e1d5edf8..92c68611 100644
--- a/libibverbs/man/CMakeLists.txt
+++ b/libibverbs/man/CMakeLists.txt
@@ -1,24 +1,18 @@
 rdma_man_pages(
-  ibv_advise_mr.3.md
   ibv_alloc_dm.3
   ibv_alloc_mw.3
-  ibv_alloc_null_mr.3.md
   ibv_alloc_parent_domain.3
   ibv_alloc_pd.3
   ibv_alloc_td.3
   ibv_asyncwatch.1
-  ibv_attach_counters_point_flow.3.md
-  ibv_attach_mcast.3.md
   ibv_bind_mw.3
   ibv_create_ah.3
   ibv_create_ah_from_wc.3
   ibv_create_comp_channel.3
-  ibv_create_counters.3.md
   ibv_create_cq.3
   ibv_create_cq_ex.3
   ibv_modify_cq.3
   ibv_create_flow.3
-  ibv_create_flow_action.3.md
   ibv_create_qp.3
   ibv_create_qp_ex.3
   ibv_create_rwq_ind_table.3
@@ -27,16 +21,8 @@ rdma_man_pages(
   ibv_create_wq.3
   ibv_devices.1
   ibv_devinfo.1
-  ibv_event_type_str.3.md
-  ibv_fork_init.3.md
   ibv_get_async_event.3
   ibv_get_cq_event.3
-  ibv_get_device_guid.3.md
-  ibv_get_device_list.3.md
-  ibv_get_device_name.3.md
-  ibv_get_pkey_index.3.md
-  ibv_get_srq_num.3.md
-  ibv_inc_rkey.3.md
   ibv_modify_qp.3
   ibv_modify_qp_rate_limit.3
   ibv_modify_srq.3
@@ -51,24 +37,15 @@ rdma_man_pages(
   ibv_post_srq_recv.3
   ibv_query_device.3
   ibv_query_device_ex.3
-  ibv_query_gid.3.md
-  ibv_query_pkey.3.md
   ibv_query_port.3
   ibv_query_qp.3
   ibv_query_rt_values_ex.3
   ibv_query_srq.3
-  ibv_rate_to_mbps.3.md
-  ibv_rate_to_mult.3.md
   ibv_rc_pingpong.1
-  ibv_read_counters.3.md
   ibv_reg_mr.3
-  ibv_req_notify_cq.3.md
-  ibv_rereg_mr.3.md
-  ibv_resize_cq.3.md
   ibv_srq_pingpong.1
   ibv_uc_pingpong.1
   ibv_ud_pingpong.1
-  ibv_wr_post.3.md
   ibv_xsrq_pingpong.1
   )
 rdma_alias_man_pages(
@@ -79,47 +56,18 @@ rdma_alias_man_pages(
   ibv_alloc_mw.3 ibv_dealloc_mw.3
   ibv_alloc_pd.3 ibv_dealloc_pd.3
   ibv_alloc_td.3 ibv_dealloc_td.3
-  ibv_attach_mcast.3 ibv_detach_mcast.3
   ibv_create_ah.3 ibv_destroy_ah.3
   ibv_create_ah_from_wc.3 ibv_init_ah_from_wc.3
   ibv_create_comp_channel.3 ibv_destroy_comp_channel.3
-  ibv_create_counters.3 ibv_destroy_counters.3
   ibv_create_cq.3 ibv_destroy_cq.3
   ibv_create_flow.3 ibv_destroy_flow.3
-  ibv_create_flow_action.3 ibv_destroy_flow_action.3
-  ibv_create_flow_action.3 ibv_modify_flow_action.3
   ibv_create_qp.3 ibv_destroy_qp.3
   ibv_create_rwq_ind_table.3 ibv_destroy_rwq_ind_table.3
   ibv_create_srq.3 ibv_destroy_srq.3
   ibv_create_wq.3 ibv_destroy_wq.3
-  ibv_event_type_str.3 ibv_node_type_str.3
-  ibv_event_type_str.3 ibv_port_state_str.3
   ibv_get_async_event.3 ibv_ack_async_event.3
   ibv_get_cq_event.3 ibv_ack_cq_events.3
-  ibv_get_device_list.3 ibv_free_device_list.3
   ibv_open_device.3 ibv_close_device.3
   ibv_open_xrcd.3 ibv_close_xrcd.3
-  ibv_rate_to_mbps.3 mbps_to_ibv_rate.3
-  ibv_rate_to_mult.3 mult_to_ibv_rate.3
   ibv_reg_mr.3 ibv_dereg_mr.3
-  ibv_wr_post.3 ibv_wr_abort.3
-  ibv_wr_post.3 ibv_wr_complete.3
-  ibv_wr_post.3 ibv_wr_start.3
-  ibv_wr_post.3 ibv_wr_atomic_cmp_swp.3
-  ibv_wr_post.3 ibv_wr_atomic_fetch_add.3
-  ibv_wr_post.3 ibv_wr_bind_mw.3
-  ibv_wr_post.3 ibv_wr_local_inv.3
-  ibv_wr_post.3 ibv_wr_rdma_read.3
-  ibv_wr_post.3 ibv_wr_rdma_write.3
-  ibv_wr_post.3 ibv_wr_rdma_write_imm.3
-  ibv_wr_post.3 ibv_wr_send.3
-  ibv_wr_post.3 ibv_wr_send_imm.3
-  ibv_wr_post.3 ibv_wr_send_inv.3
-  ibv_wr_post.3 ibv_wr_send_tso.3
-  ibv_wr_post.3 ibv_wr_set_inline_data.3
-  ibv_wr_post.3 ibv_wr_set_inline_data_list.3
-  ibv_wr_post.3 ibv_wr_set_sge.3
-  ibv_wr_post.3 ibv_wr_set_sge_list.3
-  ibv_wr_post.3 ibv_wr_set_ud_addr.3
-  ibv_wr_post.3 ibv_wr_set_xrc_srqn.3
   )
diff --git a/librdmacm/man/CMakeLists.txt b/librdmacm/man/CMakeLists.txt
index 2d1efbff..d016c926 100644
--- a/librdmacm/man/CMakeLists.txt
+++ b/librdmacm/man/CMakeLists.txt
@@ -20,7 +20,6 @@ rdma_man_pages(
   rdma_destroy_qp.3
   rdma_destroy_srq.3
   rdma_disconnect.3
-  rdma_establish.3.md
   rdma_event_str.3
   rdma_free_devices.3
   rdma_get_cm_event.3
@@ -33,7 +32,6 @@ rdma_man_pages(
   rdma_get_send_comp.3
   rdma_get_src_port.3
   rdma_getaddrinfo.3
-  rdma_init_qp_attr.3.md
   rdma_join_multicast.3
   rdma_join_multicast_ex.3
   rdma_leave_multicast.3
diff --git a/providers/efa/man/CMakeLists.txt b/providers/efa/man/CMakeLists.txt
index d6a4f57b..376f293d 100644
--- a/providers/efa/man/CMakeLists.txt
+++ b/providers/efa/man/CMakeLists.txt
@@ -1,4 +1,2 @@
 rdma_man_pages(
-  efadv_create_driver_qp.3.md
-  efadv.7.md
 )
diff --git a/providers/mlx4/man/CMakeLists.txt b/providers/mlx4/man/CMakeLists.txt
index 3df77caf..99a75101 100644
--- a/providers/mlx4/man/CMakeLists.txt
+++ b/providers/mlx4/man/CMakeLists.txt
@@ -1,6 +1,5 @@
 rdma_man_pages(
   mlx4dv_init_obj.3
   mlx4dv_query_device.3
-  mlx4dv_set_context_attr.3.md
   mlx4dv.7
 )
diff --git a/providers/mlx5/man/CMakeLists.txt b/providers/mlx5/man/CMakeLists.txt
index 0bd9eba1..876b6fcf 100644
--- a/providers/mlx5/man/CMakeLists.txt
+++ b/providers/mlx5/man/CMakeLists.txt
@@ -1,72 +1,7 @@
 rdma_man_pages(
-  mlx5dv_alloc_dm.3.md
-  mlx5dv_create_cq.3.md
-  mlx5dv_create_flow.3.md
-  mlx5dv_create_flow_action_modify_header.3.md
-  mlx5dv_create_flow_action_packet_reformat.3.md
-  mlx5dv_create_flow_matcher.3.md
-  mlx5dv_create_mkey.3.md
-  mlx5dv_create_qp.3.md
-  mlx5dv_devx_alloc_uar.3.md
-  mlx5dv_devx_create_cmd_comp.3.md
-  mlx5dv_devx_create_event_channel.3.md
-  mlx5dv_devx_get_event.3.md
-  mlx5dv_devx_obj_create.3.md
-  mlx5dv_devx_qp_modify.3.md
-  mlx5dv_devx_query_eqn.3.md
-  mlx5dv_devx_subscribe_devx_event.3.md
-  mlx5dv_devx_umem_reg.3.md
-  mlx5dv_dr_flow.3.md
-  mlx5dv_flow_action_esp.3.md
   mlx5dv_get_clock_info.3
   mlx5dv_init_obj.3
-  mlx5dv_is_supported.3.md
-  mlx5dv_open_device.3.md
   mlx5dv_query_device.3
   mlx5dv_ts_to_ns.3
-  mlx5dv_wr_post.3.md
   mlx5dv.7
 )
-rdma_alias_man_pages(
- mlx5dv_create_mkey.3 mlx5dv_destroy_mkey.3
- mlx5dv_devx_alloc_uar.3 mlx5dv_devx_free_uar.3
- mlx5dv_devx_create_cmd_comp.3 mlx5dv_devx_destroy_cmd_comp.3
- mlx5dv_devx_create_event_channel.3 mlx5dv_devx_destroy_event_channel.3
- mlx5dv_devx_create_cmd_comp.3 mlx5dv_devx_get_async_cmd_comp.3
- mlx5dv_devx_obj_create.3 mlx5dv_devx_general_cmd.3
- mlx5dv_devx_obj_create.3 mlx5dv_devx_obj_destroy.3
- mlx5dv_devx_obj_create.3 mlx5dv_devx_obj_query.3
- mlx5dv_devx_obj_create.3 mlx5dv_devx_obj_query_async.3
- mlx5dv_devx_obj_create.3 mlx5dv_devx_obj_modify.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_qp_query.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_cq_modify.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_cq_query.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_wq_modify.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_wq_query.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_srq_modify.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_srq_query.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_ind_tbl_modify.3
- mlx5dv_devx_qp_modify.3 mlx5dv_devx_ind_tbl_query.3
- mlx5dv_devx_subscribe_devx_event.3 mlx5dv_devx_subscribe_devx_event_fd.3
- mlx5dv_devx_umem_reg.3 mlx5dv_devx_umem_dereg.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_dest_table.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_dest_ibv_qp.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_dest_vport.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_flow_counter.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_drop.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_modify_header.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_packet_reformat.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_create_tag.3
- mlx5dv_dr_flow.3 mlx5dv_dr_action_destroy.3
- mlx5dv_dr_flow.3 mlx5dv_dr_domain_create.3
- mlx5dv_dr_flow.3 mlx5dv_dr_domain_destroy.3
- mlx5dv_dr_flow.3 mlx5dv_dr_domain_sync.3
- mlx5dv_dr_flow.3 mlx5dv_dr_matcher_create.3
- mlx5dv_dr_flow.3 mlx5dv_dr_matcher_destroy.3
- mlx5dv_dr_flow.3 mlx5dv_dr_rule_create.3
- mlx5dv_dr_flow.3 mlx5dv_dr_rule_destroy.3
- mlx5dv_dr_flow.3 mlx5dv_dr_table_create.3
- mlx5dv_dr_flow.3 mlx5dv_dr_table_destroy.3
- mlx5dv_wr_post.3 mlx5dv_wr_set_dc_addr.3
- mlx5dv_wr_post.3 mlx5dv_qp_ex_from_ibv_qp_ex.3
-)
-- 
2.17.1

